tests/bwc/test_partition_storage_formats.py

```diff
@@ -0,0 +1,110 @@
+import os
+import pprint
+import random
+import unittest
+
+from collections import defaultdict
+from crate.client import connect
+from crate.qa.tests import NodeProvider, insert_data, wait_for_active_shards, UpgradePath
+from pathlib import Path
+
+UPGRADE_PATH = UpgradePath('5.9.x', 'latest-nightly')
+
+def _add_data(cursor, versions, count):
+    # versions = [ old, new ]
+    cursor.executemany(
+        "INSERT INTO partitioned (version, value) VALUES (?, ?)",
+         [ (versions[i % len(versions)], i) for i in range(0, count)]
+    )
+    
+def _partition_paths(cursor):
+    paths = defaultdict(list)
+    cursor.execute('select p.values, s.path from sys.shards s, information_schema.table_partitions p where s.partition_ident = p.partition_ident;')
+    results = cursor.fetchall()
+    for (values, path) in results:
+        version = values['version']
+        paths[version].append(path)
+    pprint.pp(paths)
+    return paths
+    
+def _fdt_size(data_path):
+    size = 0
+    if not data_path:
+        return 0
+    for dirpath, dirnames, filenames in os.walk(data_path):
+        for filename in filenames:
+            path = Path(dirpath, filename)
+            ext = path.suffix.lstrip('.')
+            if ext == 'fdt':
+                filesize = path.stat().st_size
+                print(f"{path}: {filesize}")
+                size += filesize
+    return size     
+
+class PartitionStorageTest(NodeProvider, unittest.TestCase):
+    
+    def test_partition_formats_across_versions(self):
+        with self.subTest(repr(UPGRADE_PATH)):
+            try:
+                self.setUp()
+                self._run_tests(UPGRADE_PATH, nodes=3)
+            finally:
+                self.tearDown()       
+        
+    def _run_tests(self, upgrade_path, nodes):
+        
+        # start initial cluster
+        settings = {
+            "transport.netty.worker_count": 16
+        }
```

Comment:
Afaik we set the worker count in the other bwc tests due to some bug in an older version. Shouldn't be necessary to set this here.